modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -568,7 +500,7 @@ GrainTracker::assignGrains()
       grain._status = Status::MARKED;                 // Mark the grain
 
     // Set up the first reserve grain index based on the largest grain ID
-    _reserve_grain_first_idx = _max_curr_grain_id + 1;
+    _reserve_grain_first_index = _max_curr_grain_id + 1;
```

Comment:
This is actually correct... I've thought about this more than I care to admit. Once you realize that the GrainTracker only works with contiguous IDs it makes sense. The "starting" reserve index (as in the actual position in the vector) will then corresponding to the largest ID plus one. 

That's a lot of what this PR was supposed to fix. I went through and carefully and looked at every use of ID and index and I'm fairly certain I've fixed them all up, at least that's what I was trying to do.